server/src/main/java/org/elasticsearch/common/regex/Regex.java

```diff
@@ -44,6 +44,13 @@ public static boolean isSimpleMatchPattern(String str) {
         return str.indexOf('*') != -1;
     }
 
+    /**
+     * Is the str a lucene type of pattern
+     */
+    public static boolean isLuceneRegex(String str) {
+        return str.length() > 1 && str.charAt(0) == '/' && str.charAt(str.length() - 1) == '/';
```

Comment:
I don't think this "regex" style is used outside of security, so I don't think we need to add this method to `server/`
Maybe `Automatons` would be better?

Reply:
Yes, I think it's better, thanks.